Update docs and miscellaneous (#3)
fortinet/fortigate-autoscale-aws: README.md

# FortiGate Autoscale on AWS

> The implementation for FortiGate Autoscale using [Fortinet Autoscale](https://github.com/fortinet/autoscale-core) on Amazon AWS platfrom.

Author: ```suggestion
The implementation for FortiGate Autoscale using Fortinet [Autoscale Core](https://github.com/fortinet/autoscale-core) on Amazon AWS platfrom.
```
@klc178 please review this change.

Reviewer: ```suggestion
An implementation for FortiGate Autoscale for the AWS SDK platform API with a Dynamo DB storage backend. This implementation requires Fortinet [Autoscale Core](https://github.com/fortinet/autoscale-core).
```

> This project provides a multi-group Auto Scaling functionality for Fortinet FortiGate EC2 instances to form an HA cluster with failover protection.

Reviewer: ```suggestion
This project provides multi-group Auto Scaling functionality for Fortinet FortiGate EC2 instances to form an HA cluster with failover protection.
```

This project has the following features:

> 1. Multi-group Hybrid Licensing models:

Reviewer: ```suggestion
1. Multi-group Licensing models:
```

Author: as discussed, we will keep the Hybrid so we do no take this suggestion.

>    1. **Hybrid**: 1 fix-sized Auto Scaling group of 2 (and more) BYOL FortiGate Instances, and 1 dynamically scalable Auto Scaling group of (0 or more) On-Demand FortiGate Instances.

Reviewer: ```suggestion
   1. **BYOL-Only**: 1 dynamically scalable Auto Scaling group of (0 or more) Bring Your Own Licence (BYOL) FortiGate instances.
```

>    2. **BYOL-Only**: 1 dynamically scalable Auto Scaling group of (0 or more) BYOL FortiGate Instances.

Reviewer: ```suggestion
   3. **PAYG-Only**: 1 dynamically scalable Auto Scaling group of (0 or more) On-Demand FortiGate instances.
```

>    3. **PAYG-Only**: 1 dynamically scalable Auto Scaling group of (0 or more) On-Demand FortiGate Instances.

Reviewer: ```suggestion
   1. **Hybrid**: 1 fix-sized Auto Scaling group of 2 (and more) BYOL FortiGate instances, and 1 dynamically scalable Auto Scaling group of (0 or more) On-Demand FortiGate instances.
```

2. AWS Transit Gateway Integration.

## Deployment packages

> 

Reviewer: ```suggestion
To generate local deployment packages:
```

> 

Reviewer: ```suggestion
To obtain the deployment package, use one of the following:
```

>   * Download the **aws-cloudformation.zip** for the latest production version from [project release page](https://github.com/fortinet/fortigate-autoscale-aws/releases).

Reviewer: ```suggestion
  *  From the [project release page](https://github.com/fortinet/fortigate-autoscale-aws/releases), download the **aws-cloudformation.zip** for the latest production version.
```

>   * Download the **aws-cloudformation.zip** for the latest production version from [project release page](https://github.com/fortinet/fortigate-autoscale-aws/releases).

Reviewer: I approve.
  * Download the package **aws-cloudformation.zip** directly from the [project release page](https://github.com/fortinet/fortigate-autoscale-aws/releases).

>   * or, manually generate the deployment package:

Reviewer: ```suggestion
  * Manually generate the deployment package:
```

>   * or, manually generate the deployment package:

Reviewer: Confirmed:
  * Manually generate the deployment package in your local workspace:

>     1. Chechkout the **main** branch of the project.

Reviewer: ```suggestion
    1. Check out the **main** branch of the project.
```

Reviewer: approved.
    1. From the [project release page](https://github.com/fortinet/fortigate-autoscale-aws/releases), download the source code (.zip or .tar.gz) for the latest version.

    2. Run `npm run build-artifacts` at the project root directory.

>     3. Deployment packages **aws-cloudformation.zip** will be available in the **dist/artifacts** directory.

Reviewer: ``suggestion
    3. The deployment package **aws-cloudformation.zip** will be available in the **dist/artifacts** directory.
```

Author: accepted down here.

>     3. Deployment packages **aws-cloudformation.zip** will be available in the **dist/artifacts** directory.

Reviewer: looks great! (steps 2 -5)

> ## Deployment guides

Reviewer: ```suggestion
## Deployment guide
```

Deployment guides are available from the Fortinet Document Library:

>   + [<NAME_PLACE_HOLDER>](<LINK_PLACE_HOLDER>)

Author: @klc178 please update it here with the doc reference to the official documentation.

Reviewer: ```suggestion
  + [ FortiGate / FortiOS 6.2 Deploying auto scaling on AWS](https://docs.fortinet.com/vm/aws/fortigate/6.2/aws-cookbook/6.2.0/397979/deploying-auto-scaling-on-aws)
    + [ FortiGate / FortiOS 6.4 Deploying auto scaling on AWS](https://docs.fortinet.com/vm/aws/fortigate/6.4/aws-cookbook/6.4.0/397979/deploying-auto-scaling-on-aws)
```

Reviewer: are we supporting 6.4? It isn't in the drop down.

Author: > are we supporting 6.4? It isn't in the drop down.

Not in this version but in the plan to support it soon.

> Deployment guides are available from the Fortinet Document Library:

Reviewer: ```suggestion
A deployment guide is available from the Fortinet Document Library:
```

## Launch a demo

> | Non-TGW (new VPC) | Non-TGW (existing VPC) | TGW Integration (new VPC) |

Reviewer: ```suggestion
| New VPC, no-TGW | Eexisting VPC, no-TGW | TGW Integration (new VPC) |
```

> | New VPC, no-TGW | Eexisting VPC, no-TGW | TGW Integration (new VPC) |

Reviewer: ```suggestion
| New VPC, no-TGW | Existing VPC, no-TGW | TGW Integration (new VPC) |
```

# Support
Fortinet-provided scripts in this and other GitHub projects do not fall under the regular Fortinet technical support scope and are not supported by FortiCare Support Services.

> For direct issues, please refer to the [Issues](https://github.com/fortinet/fortigate-autoscale/issues) tab of this GitHub project.

Author: ```suggestion
For direct issues, please refer to the [Issues](https://github.com/fortinet/fortigate-autoscale-aws/issues) tab of this GitHub project.
```
@klc178 please confirm.

Reviewer: confirmed for:
For direct issues, please refer to the [Issues](https://github.com/fortinet/fortigate-autoscale-aws/issues) tab of this GitHub project.

For other questions related to this project, contact [github@fortinet.com](mailto:github@fortinet.com).

## License

> [License](https://github.com/fortinet/fortigate-autoscale/blob/master/LICENSE) © Fortinet Technologies. All rights reserved.

Author: ```suggestion
[License](https://github.com/fortinet/fortigate-autoscale-aws/blob/master/LICENSE) © Fortinet Technologies. All rights reserved.
```
@klc178 please confirm.

Reviewer: Confirmed:
[License](https://github.com/fortinet/fortigate-autoscale-aws/blob/master/LICENSE) © Fortinet Technologies. All rights reserved.

Reviewer: Please update the date in the license file to 2020.
